File: app/src/main/java/dev/ragnarok/fenrir/db/model/entity/UserEntity.java
package dev.ragnarok.fenrir.db.model.entity;


import dev.ragnarok.fenrir.model.Sex;
import dev.ragnarok.fenrir.model.UserPlatform;

public class UserEntity {

    private final int id;

    private String firstName;

    private String lastName;

    private boolean online;

    private boolean onlineMobile;

    private int onlineApp;

    private String photo50;

    private String photo100;

    private String photo200;

    private String photoMax;

    private long lastSeen;

    private @UserPlatform
    int platform;

    private String status;

    private @Sex
    int sex;

    private String domain;

    private String maiden_name;

    private boolean friend;
    private int friendStatus;
    private boolean can_write_private_message;
    private boolean blacklisted_by_me;
    private boolean blacklisted;
    private boolean verified;
    private boolean can_access_closed;

    public UserEntity(int id) {
        this.id = id;
    }

    public int getId() {
        return id;
    }

    public String getFirstName() {
        return firstName;
    }

    public UserEntity setFirstName(String firstName) {
        this.firstName = firstName;
        return this;
    }

    public String getLastName() {
        return lastName;
    }

    public UserEntity setLastName(String lastName) {
        this.lastName = lastName;
        return this;
    }

    public boolean isOnline() {
        return online;
    }

    public UserEntity setOnline(boolean online) {
        this.online = online;
        return this;
    }

    public boolean isOnlineMobile() {
        return onlineMobile;
    }

    public UserEntity setOnlineMobile(boolean onlineMobile) {
        this.onlineMobile = onlineMobile;
        return this;
    }

    public int getOnlineApp() {
        return onlineApp;
    }

    public UserEntity setOnlineApp(int onlineApp) {
        this.onlineApp = onlineApp;
        return this;
    }

    public String getPhoto50() {
        return photo50;
    }

    public UserEntity setPhoto50(String photo50) {
        this.photo50 = photo50;
        return this;
    }

    public String getPhoto100() {
        return photo100;
    }

    public UserEntity setPhoto100(String photo100) {
        this.photo100 = photo100;
        return this;
    }

    public String getPhoto200() {
        return photo200;
    }

    public UserEntity setPhoto200(String photo200) {
        this.photo200 = photo200;
        return this;
    }

    public long getLastSeen() {
        return lastSeen;
    }

    public UserEntity setLastSeen(long lastSeen) {
        this.lastSeen = lastSeen;
        return this;
    }

    public @UserPlatform
    int getPlatform() {
        return platform;
    }

    public UserEntity setPlatform(@UserPlatform int platform) {
        this.platform = platform;
        return this;
    }

    public String getStatus() {
        return status;
    }

    public UserEntity setStatus(String status) {
        this.status = status;
        return this;
    }

    public @Sex
    int getSex() {
        return sex;
    }

    public UserEntity setSex(@Sex int sex) {
        this.sex = sex;
        return this;
    }

    public String getDomain() {
        return domain;
    }

    public UserEntity setDomain(String domain) {
        this.domain = domain;
        return this;
    }

    public boolean isFriend() {
        return friend;
    }

    public UserEntity setFriend(boolean friend) {
        this.friend = friend;
        return this;
    }

    public String getPhotoMax() {
        return photoMax;
    }

    public UserEntity setPhotoMax(String photoMax) {
        this.photoMax = photoMax;
        return this;
    }

    public boolean getCanWritePrivateMessage() {
        return can_write_private_message;
    }

    public UserEntity setCanWritePrivateMessage(boolean can_write_private_message) {
        this.can_write_private_message = can_write_private_message;
        return this;
    }

    public boolean getBlacklisted_by_me() {
        return blacklisted_by_me;
    }

    public UserEntity setBlacklisted_by_me(boolean blacklisted_by_me) {
        this.blacklisted_by_me = blacklisted_by_me;
        return this;
    }

    public boolean getBlacklisted() {
        return blacklisted;
    }

    public UserEntity setBlacklisted(boolean blacklisted) {
        this.blacklisted = blacklisted;
        return this;
    }

    public int getFriendStatus() {
        return friendStatus;
    }

    public UserEntity setFriendStatus(int friendStatus) {
        this.friendStatus = friendStatus;
        return this;
    }

    public boolean isVerified() {
        return verified;
    }

    public UserEntity setVerified(boolean verified) {
        this.verified = verified;
        return this;
    }

    public boolean isCan_access_closed() {
        return can_access_closed;
    }

    public UserEntity setCan_access_closed(boolean can_access_closed) {
        this.can_access_closed = can_access_closed;
        return this;
    }

    public String getMaiden_name() {
        return maiden_name;
    }

    public UserEntity setMaiden_name(String maiden_name) {
        this.maiden_name = maiden_name;
        return this;
    }
}

File: app_fenrir/src/main/kotlin/dev/ragnarok/fenrir/db/model/entity/UserEntity.kt
package dev.ragnarok.fenrir.db.model.entity

import dev.ragnarok.fenrir.model.Sex
import dev.ragnarok.fenrir.model.UserPlatform

class UserEntity(val id: Int) {
    var firstName: String? = null
        private set
    var lastName: String? = null
        private set
    var isOnline = false
        private set
    var isOnlineMobile = false
        private set
    var onlineApp = 0
        private set
    var photo50: String? = null
        private set
    var photo100: String? = null
        private set
    var photo200: String? = null
        private set
    var photoMax: String? = null
        private set
    var lastSeen: Long = 0
        private set
    var bdate: String? = null
        private set

    @get:UserPlatform
    @UserPlatform
    var platform = 0
        private set
    var status: String? = null
        private set

    @get:Sex
    @Sex
    var sex = 0
        private set
    var domain: String? = null
        private set
    var maiden_name: String? = null
        private set
    var isFriend = false
        private set
    var friendStatus = 0
        private set
    var canWritePrivateMessage = false
        private set
    var blacklisted_by_me = false
        private set
    var blacklisted = false
        private set
    var isVerified = false
        private set
    var isCan_access_closed = false
        private set

    fun setBdate(bdate: String?): UserEntity {
        this.bdate = bdate
        return this
    }

    fun setFirstName(firstName: String?): UserEntity {
        this.firstName = firstName
        return this
    }

    fun setLastName(lastName: String?): UserEntity {
        this.lastName = lastName
        return this
    }

    fun setOnline(online: Boolean): UserEntity {
        isOnline = online
        return this
    }

    fun setOnlineMobile(onlineMobile: Boolean): UserEntity {
        isOnlineMobile = onlineMobile
        return this
    }

    fun setOnlineApp(onlineApp: Int): UserEntity {
        this.onlineApp = onlineApp
        return this
    }

    fun setPhoto50(photo50: String?): UserEntity {
        this.photo50 = photo50
        return this
    }

    fun setPhoto100(photo100: String?): UserEntity {
        this.photo100 = photo100
        return this
    }

    fun setPhoto200(photo200: String?): UserEntity {
        this.photo200 = photo200
        return this
    }

    fun setLastSeen(lastSeen: Long): UserEntity {
        this.lastSeen = lastSeen
        return this
    }

    fun setPlatform(@UserPlatform platform: Int): UserEntity {
        this.platform = platform
        return this
    }

    fun setStatus(status: String?): UserEntity {
        this.status = status
        return this
    }

    fun setSex(@Sex sex: Int): UserEntity {
        this.sex = sex
        return this
    }

    fun setDomain(domain: String?): UserEntity {
        this.domain = domain
        return this
    }

    fun setFriend(friend: Boolean): UserEntity {
        isFriend = friend
        return this
    }

    fun setPhotoMax(photoMax: String?): UserEntity {
        this.photoMax = photoMax
        return this
    }

    fun setCanWritePrivateMessage(can_write_private_message: Boolean): UserEntity {
        canWritePrivateMessage = can_write_private_message
        return this
    }

    fun setBlacklisted_by_me(blacklisted_by_me: Boolean): UserEntity {
        this.blacklisted_by_me = blacklisted_by_me
        return this
    }

    fun setBlacklisted(blacklisted: Boolean): UserEntity {
        this.blacklisted = blacklisted
        return this
    }

    fun setFriendStatus(friendStatus: Int): UserEntity {
        this.friendStatus = friendStatus
        return this
    }

    fun setVerified(verified: Boolean): UserEntity {
        isVerified = verified
        return this
    }

    fun setCan_access_closed(can_access_closed: Boolean): UserEntity {
        isCan_access_closed = can_access_closed
        return this
    }

    fun setMaiden_name(maiden_name: String?): UserEntity {
        this.maiden_name = maiden_name
        return this
    }
}